### PR TITLE
Pass `uploadToServer` option to windows crash reporter

### PR DESCRIPTION
**Pull request by tarruda**
Note that `get/setUploadToServer` is not implemented because the CustomInfo structure is only set when `start` is called, so to change the value of `uploadToServer` the user needs to call `start` again(which is how it already works for most options).

### CODE REVIEW

File: lib/common/api/crash-reporter.js [hunk 1]
```diff
@@ -43,7 +43,12 @@ class CrashReporter {
       throw new Error('companyName is a required option to crashReporter.start')
     }
     if (submitURL == null) {
-      throw new Error('submitURL is a required option to crashReporter.start')
+      if (uploadToServer) {
+        throw new Error('submitURL is a required option to crashReporter.start')
+      } else {
+        // use a dummy upload URL
+        submitURL = 'http://dummy/upload/url'
```
Comment (kevinsawicki) [inline lib/common/api/crash-reporter.js]: Hmm, so if you call start with `uploadToServer` with `false` and no `submitURL` and then later call `setUploadToServer(true)` and then get a crash, you'll see your machine make a request to this URL right?

Would an empty submit URL be possible to use instead? Or does that cause issues somewhere else?

Comment (tarruda) [inline lib/common/api/crash-reporter.js]: > later call setUploadToServer(true) and then get a crash, you'll see your machine make a request to this URL right

Yes, I considered using `http://localhost` but it is possible that the user has a local http server.

**Comment by kevinsawicki** [comment]
Thanks for adding support for this @tarruda 👍 🏁

**Comment by kevinsawicki** [comment]
Just got a flaky looking failure here on AppVeyor:

```
not ok 180 crashReporter module without sandbox should not send minidump if uploadToServer is false
  Error: EBUSY: resource busy or locked, unlink 'C:\Users\appveyor\AppData\Local\Temp\1\electronCrashReporterSpec-11733-1148-1priygi.7pgidaemi\Zombies Crashes\a92fbc58-7a87-4fd3-8012-3c70de2df434.dmp'
      at Object.fs.unlinkSync (fs.js:1007:18)
      at testDone (C:\projects\electron\spec\api-crash-reporter-spec.js:96:14)
```

https://ci.appveyor.com/project/electron-bot/electron/build/244/job/rp59bmb2epvvbh52#L1518

Perhaps we should use `graceful-fs` or ignore delete errors?

**Comment by tarruda** [comment]
> Perhaps we should use graceful-fs or ignore delete errors?

The file is deleted is to ensure the crash dump is created(and not uploaded), my guess is that the crash reporter was still writing to the file when `unlink` was called. The reason I explicitly `unlink` the dump is because of some differences in how crashReporter directories are handled(it seems on mac they are initialized at startup and not changed later?), causing dumps from other tests to leak into this test.

In any case, I think it is better to not depend on `unlink` to know that the dump was created. I will send a PR to fix this soon.

**Pull request merged**